pallets/collective-proxy/src/lib.rs

```diff
@@ -121,5 +179,68 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Register a proxy account for the sender that is able to make calls on its behalf.
+        ///
+        /// The dispatch origin for this call must be _Signed_.
+        ///
+        /// Parameters:
+        /// - `proxy`: The account that the `caller` would like to make a proxy.
+        /// - `filter`: Call filter used for the proxy
+        #[pallet::call_index(1)]
+        #[pallet::weight(T::WeightInfo::add_proxy(T::MaxProxies::get()))]
+        pub fn add_proxy(
```

Comment:
What happens when the **ProxyAdmin** tries to overwrite an existing proxy with a stricter filter? (e.g., temporary restriction during maintenance/emergency)

Reply:
Reworked to map